packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -196,6 +167,22 @@ export class AssociationManager {
     }
   }
 
+  public static associateTagOption(resource: cdk.IResource, resourceId: string, tagOption: ITagOption): void {
+    const tagAssocationConstructId = `TagOptionAssociation${hashValues(resource.node.addr, tagOption.node.addr)}`;
+    new CfnTagOptionAssociation(resource as cdk.Resource, tagAssocationConstructId, {
+      resourceId: resourceId,
+      tagOptionId: tagOption.tagOptionId,
+    });
+  }
+
+  public static associateTagOptions(resource: cdk.IResource, resourceId: string, tagOptions: TagOptions): void {
+    for (const [_, tagOptionValuesForKey] of Object.entries(tagOptions.tagOptionsMap)) {
```

Comment:
This is probably terrible, and maybe we could make a method that just returns a flattened list or easier way to enumerate the underlying tag options